include/logger.h

```diff
@@ -58,12 +68,15 @@ template <const char *Fmt> class Record {
 
 public:
   Record(std::ostream &s, ContextInfo input_info, std::mutex &Logging_lock)
-      : stream(s), info(input_info), lock(Logging_lock) {
+      : stream(s), info(input_info), lock(Logging_lock), hash(0) {
```

Comment:
Could you re-describe some of the uses of the hash again? I remember we talked about this earlier but it's sort of hazy...